pkg/provider/consulcatalog/config_test.go

```diff
@@ -3206,3 +3206,448 @@ func extractNSFromProvider(providers []*Provider) []string {
 	}
 	return res
 }
+
+func TestFilterHealthStatuses(t *testing.T) {
+	testCases := []struct {
+		desc         string
+		items        []itemData
+		defaultRule  string
+		expected     *dynamic.Configuration
+		strictChecks []string
```

Comment:
nit: could you move `strictChecks` bellow `items` so we can more easily see the configuration of the test?